### PR TITLE
Fix xgboost label inv in binary settings

### PR DESCRIPTION
**Pull request by pat-s**
See https://github.com/mlr-org/mlr3learners/issues/32

cc @001ben

### CODE REVIEW

File: R/RLearner_classif_xgboost.R [hunk 1]
```diff
@@ -63,9 +63,10 @@ trainLearner.classif.xgboost = function(.learner, .task, .subset, .weights = NUL
   td = getTaskDesc(.task)
   parlist = list(...)
   nc = length(td$class.levels)
+  nlvls = length(td$class.levels)
```
Comment (mllg) [inline R/RLearner_classif_xgboost.R]: we don't need this twice.

File: R/RLearner_classif_xgboost.R [hunk 2]
```diff
@@ -79,6 +80,10 @@ trainLearner.classif.xgboost = function(.learner, .task, .subset, .weights = NUL
 
   task.data = getTaskData(.task, .subset, target.extra = TRUE)
   label = match(as.character(task.data$target), td$class.levels) - 1
+
+  # recode to 0:1 to that for the binary case the positive class translates to 1 (https://github.com/mlr-org/mlr3learners/issues/32)
+  # task.data$target is guaranteed to have the factor levels in the right order
```
Comment (mllg) [inline R/RLearner_classif_xgboost.R]: have you doubled checked that this holds for mlr?

File: R/RLearner_classif_xgboost.R (hunk 1 above)
Comment (pat-s) [reply R/RLearner_classif_xgboost.R]: Ups :+1:

File: R/RLearner_classif_xgboost.R (hunk 2 above)
Comment (pat-s) [reply R/RLearner_classif_xgboost.R]: Yes. Also the tests should ensure this.

**Comment by pat-s** [comment]
@berndbischl @mllg 

The error we're getting in the tests is really weird. I see no change that would affect a change of probs in the following way:

```r
── 1. Failure: classif_xgboost (@test_classif_xgboost.R#58)  ───────────────────

`p` not equal to `old.probs`.

87/107 mismatches (average diff: 2.98e-08)

[1] 0.371 - 0.371 == 2.98e-08

[2] 0.371 - 0.371 == 2.98e-08

[3] 0.371 - 0.371 == 2.98e-08

[4] 0.371 - 0.371 == 2.98e-08

[5] 0.371 - 0.371 == 2.98e-08

[6] 0.371 - 0.371 == 2.98e-08

[7] 0.371 - 0.371 == 2.98e-08

[8] 0.371 - 0.371 == 2.98e-08

[9] 0.371 - 0.371 == 2.98e-08
```

Would you mind taking a look? Thanks.

**Comment by pat-s** [comment]
I am not quite sure what's going on currently with all the tolerance problems (see also #2667) but other than that I'd say we're ready to merge.

@mllg

**Pull request merged**